helpdesk/models.py

```diff
@@ -1099,6 +1108,8 @@ def attachment_path(self, filename):
         if settings.DEFAULT_FILE_STORAGE == "django.core.files.storage.FileSystemStorage":
             if not os.path.exists(att_path):
                 os.makedirs(att_path, 0o777)
+        if settings.USE_S3 is True:
+            self.get_upload_path(filename)
```

Comment:
```suggestion
            return default_storage.get_available_name(att_path)
```

You are building the path twice - above this as `att_path` and in `get_upload_path`. Rather pass `att_path` to the function and just use `s.get_available_name(att_path)` instead of recreating the path